module/state/src/commonTest/kotlin/org/cru/godtools/tool/state/StateTest.kt

```diff
@@ -40,52 +36,32 @@ class StateTest {
     @Test
     fun testChangeFlow() = runTest {
         var i = 0
-        val channel = Channel<Int>()
-        val flow = state.changeFlow(KEY, KEY2) { i++ }
-            .onEach { channel.send(it) }
-            .launchIn(this)
+        state.changeFlow(KEY, KEY2) { i++ }.test {
+            // initial value
+            assertEquals(0, awaitItem())
 
-        // initial value
-        assertEquals(0, channel.receive(500))
+            // update state for monitored key
+            state[KEY] = "a"
+            assertEquals(1, awaitItem())
 
-        // update state for monitored key
-        state[KEY] = "a"
-        assertEquals(1, channel.receive(500))
+            // update state for other monitored key
+            state[KEY2] = "a"
+            assertEquals(2, awaitItem())
 
-        // update state for other monitored key
-        state[KEY2] = "a"
-        assertEquals(2, channel.receive(500))
-
-        // update state for a different key
-        state["other$KEY"] = "a"
-        delay(100)
-        assertTrue(channel.isEmpty)
-
-        // shut down flow
-        flow.cancel()
+            // update state for a different key
+            state["other$KEY"] = "a"
+            expectNoEvents()
+        }
         assertEquals(3, i)
     }
 
     @Test
     fun testChangeFlowNoKeys() = runTest {
         var count = 0
-        val channel = Channel<Int>()
-        val flow = state.changeFlow { count++ }
-            .onEach { channel.send(it) }
-            .launchIn(this)
-
-        // initial value
-        assertEquals(0, channel.receive(500))
-
-        // update state for multiple keys, should never emit a new value
-        for (i in 1..10) {
-            state["$KEY$i"] = "a"
-            delay(100)
-            assertTrue(channel.isEmpty)
+        state.changeFlow { count++ }.test {
+            assertEquals(0, awaitItem())
+            awaitComplete()
```

Comment:
Seems we lost the test case that all subsequent values are empty, but maybe not a big deal.

Reply:
`awaitComplete()` means that no more values will be emitted by the flow, so updating any subsequent values won't be able to emit anything even if it did change the state.

If the flow didn't naturally complete after the initial emission then it would make sense to test changing random values.

Reply:
To be fair, this is testing implementation behavior specifically and not really testing the intent of the last value emitted never changing to true.

There are a few other Turbine testing methods that might avoid this being implementation specific